guides/source/active_record_querying.md

```diff
@@ -1935,6 +1935,61 @@ SELECT books.* FROM books WHERE books.out_of_print = true
 
 [`unscoped`]: https://api.rubyonrails.org/classes/ActiveRecord/Scoping/Default/ClassMethods.html#method-i-unscoped
 
+### New Chains Inside Scope Block
+
+Unlike class methods, [`scope`][] can easily start a new clean chain against the
+model it is defined on.
+
+```ruby
+class Topic < ApplicationRecord
+  scope :toplevel, -> { where(parent_id: nil) }
+  scope :children, -> { where.not(parent_id: nil) }
+  scope :has_children, -> {
+    where(id: Topic.children.select(:parent_id))
+  }
+end
+
+Topic.toplevel.has_children
+```
+
+Inside `has_children` the `Topic` chain generates a subquery like this:
```

Comment:
What is a chain?

Reply:
Maybe "chained relation" would be better?

Reply:
I'm not sure, I'm confused with what this section is documenting. Maybe change the documentation to explain it better? As it is right now, even I don't understand what is going on here.

For example I believe the `where` inside the `has_children` scope will respect the `toplevel` scope, but the `Topic.children` will not. I'd document that explaining why `where` is different from `Topic.where` or `self.where`. It seems the NOTE is trying to explain that, but it isn't very clean and don't explain the differences.

Let's revert this PR and work in improving it.

Reply:
Both `where` and `Topic.children` will respect the `toplevel` scope in the example. Modify the script from #44792 like so:

```rb
  def self.toplevel
    where('"toplevel" = "toplevel"')
  end

  . . .

  def self.has_children
    puts '-------------------------------'
    puts where("1=1").to_sql
    puts self.where("1=1").to_sql
    puts Topic.all.to_sql
    puts Topic.children.to_sql
    puts '-------------------------------'
  end
```

You will see `"toplevel" = "toplevel"` included in the generated SQL for each.

I was trying to stay close to the original example from https://github.com/rails/rails/pull/32380 but perhaps revising it so that the class method is reiterated in the query output would help?

"chain" comes from matthewd's explanation here: https://github.com/rails/rails/issues/44792#issuecomment-1663929690 I can avoid it though and use "chained relation" instead per @adrianna-chang-shopify's suggestion.